gobblin-core/src/main/java/gobblin/converter/initializer/ConverterInitializerFactory.java

```diff
@@ -68,15 +70,27 @@ private static ConverterInitializer newInstance(State state, WorkUnitStream work
     }
 
     List<ConverterInitializer> cis = Lists.newArrayList();
-    JdbcWriterCommandsFactory factory = new JdbcWriterCommandsFactory();
     for (String converterClass : converterClasses) {
-      if (AvroToJdbcEntryConverter.class.getName().equals(converterClass)) {
-        if (workUnits.isSafeToMaterialize()) {
-          cis.add(new AvroToJdbcEntryConverterInitializer(state, workUnits.getMaterializedWorkUnitCollection(),
-              factory, branches, branchId));
-        } else {
-          throw new RuntimeException(AvroToJdbcEntryConverter.class.getName() + " does not support work unit streams.");
-        }
+      Class<?> klazz;
+      try {
```

Comment:
Good point. I'll add the method `getInitializer` to the whole `Converter` or `DataWriterBuilder` type hierarchy.